extension/app/src/context/ProtectedRoute.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Button,
+  ExternalLinkIcon,
+  LogoHorizontalColorLogo,
+  LogoutIcon,
+  Spinner,
+} from "@dust-tt/sparkle";
+import type { ReactNode } from "react";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { useAuth } from "./AuthProvider";
+
+export const ProtectedRoute = ({ children }: { children: ReactNode }) => {
+  const { isLoading, isAuthenticated, handleLogout } = useAuth();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!isLoading && !isAuthenticated) {
+      navigate("/login");
+    }
+  }, [isLoading, isAuthenticated, navigate]);
+
+  if (isLoading) {
+    return (
+      <div className="flex flex-col p-4 gap-2 h-screen">
+        <div className="flex justify-center items-center w-full h-full">
+          <Spinner />
+        </div>
+      </div>
+    );
+  }
+
+  return (
```

Comment:
Protected route renders the header with logout button, can be controversial -> we can keep as is, or rename the component to clarify that it also renders the header, or I create another component for the header but I seems overkill to me